### DISCUSSION TITLE
Android -- No traceparent in HTTP headers?

### DISCUSSION THREAD

**Discussion by istashak** [open-telemetry/opentelemetry-java-instrumentation#10790] [Q&A]
I'm working on a kotlin SDK, and I'm currently using the call factory to insert telemetry headers into the HTTP requests:

```
import io.opentelemetry.instrumentation.okhttp.v3_0.OkHttpTelemetry

...

OkHttpTelemetry.builder(openTelemetry).build().newCallFactory(createClient())
```

My unit tests all pass in the SDK source, and I can see the traceparent headers being added there... But, when I inspect the outgoing requests in Android studio for our test Android client the headers are missing. I can see the call stack for the request, and the call to the io.opentelemetry.instrumentation.okhttp.v3_0.TracingCallFactory is being made, but I get no traceparent.

Can someone please suggest what I might do to remedy this?

**Comment by trask** [comment]
cc @open-telemetry/android-approvers in case they have any ideas

(also if you haven't already, check out https://github.com/open-telemetry/opentelemetry-android)

**Reply by istashak** [reply]
Thanks for the reply. Since I'm developing an SDK I was hoping to use a solution that worked agnostically across a multitude of client types.

**Comment by laurit** [comment]
Headers should be injected in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/b78316b8d4df5d3d4a6ab099c9a75208f4fe3636/instrumentation/okhttp/okhttp-3.0/library/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/internal/TracingInterceptor.java#L60 Use a debugger to see what is going on. Maybe you are using the noop instance of OpenTelemetry?

**Reply by breedx-splk** [reply]
Yeah, and maybe we can get a little more code context?
